src/runtime/StateSerialization/RuntimeData.cs

```diff
@@ -17,7 +17,27 @@ namespace Python.Runtime
 {
     public static class RuntimeData
     {
-        private static Type? _formatterType;
+
+        private readonly static Func<IFormatter> DefaultFormatter = () => new BinaryFormatter();
```

Comment:
I know that this is a significant change in behaviour, but shouldn't we default to Noop here? The feature is rarley used (in particular, it's never used when .NET is embedded into Python or if the runtime is never unloaded).

Reply:
I think that it is a good idea, but I'm not sure it is within scope of that PR.

Reply:
I think the confusion is in the first comment "offer other fixes for issues 2335, 2221, 2282". Maybe the comment should corrected to exclude 2335 and 2282 regarding the BinaryFormatter in dotnet 8?

Reply:
@psyfrett My bad, I incorrectly worded things here, it should have been "offers another workaround for issues ...". I edited the description.